c_glib/gandiva-glib/node.cpp

```diff
@@ -395,42 +434,1438 @@ ggandiva_function_node_get_parameters(GGandivaFunctionNode *node)
   return priv->parameters;
 }
 
-G_END_DECLS
 
-std::shared_ptr<gandiva::Node>
-ggandiva_node_get_raw(GGandivaNode *node)
+G_DEFINE_TYPE(GGandivaLiteralNode,
+              ggandiva_literal_node,
+              GGANDIVA_TYPE_NODE)
+
+static void
+ggandiva_literal_node_init(GGandivaLiteralNode *literal_node)
 {
-  auto priv = GGANDIVA_NODE_GET_PRIVATE(node);
-  return priv->node;
 }
 
-GGandivaFieldNode *
-ggandiva_field_node_new_raw(std::shared_ptr<gandiva::Node> *gandiva_node,
-                            GArrowField *field)
+static void
+ggandiva_literal_node_class_init(GGandivaLiteralNodeClass *klass)
 {
-  auto field_node = g_object_new(GGANDIVA_TYPE_FIELD_NODE,
-                                 "node", gandiva_node,
-                                 "field", field,
-                                 NULL);
-  return GGANDIVA_FIELD_NODE(field_node);
 }
 
-GGandivaFunctionNode *
-ggandiva_function_node_new_raw(std::shared_ptr<gandiva::Node> *gandiva_node,
-                               const gchar *name,
-                               GList *parameters,
-                               GArrowDataType *return_type)
+
+typedef struct GGandivaBooleanLiteralNodePrivate_ {
+  gboolean is_true;
+} GGandivaBooleanLiteralNodePrivate;
```

Comment:
Can we implement the original raw value getter without property? It seems that property is overdone for this case.

For example:

```cpp
template <typename Type>
return Type
ggandiva_literal_node_get(GGandivaLiteralNode *node)
{
  auto gandiva_node = ggandiva_node_get_raw(GGANDIVA_NODE(node));
  return boost::get<Type>(gandiva_node->holder());
}

return gboolean
ggandiva_boolean_literal_node_get_value(GGandivaBooleanLiteralNode *node)
{
  auto value = ggandiva_literal_node_get<bool>(GGANDIVA_LITERAL_NODE(node));
  return static_cast<gboolean>(value);
}

return guint8
ggandiva_uint8_literal_node_get_value(GGandivaUInt8LiteralNode *node)
{
  return ggandiva_literal_node_get<uint8_t>(GGANDIVA_LITERAL_NODE(node));
}
```

Reply:
It couldn’t access `holder()`.

```cpp
../gandiva-glib/node.cpp:36:39: error: member access into incomplete type 'std::__1::shared_ptr<gandiva::Node>::element_type' (aka 'gandiva::Node')
  return boost::get<Type>(gandiva_node->holder());
```

I’ve tried the following example to access by `LiteralNode`.

```cpp
template <typename Type>
Type
ggandiva_literal_node_get(GGandivaLiteralNode *node)
{
  auto gandiva_literal_node = 
    dynamic_cast<gandiva::LiteralNode*>(ggandiva_node_get_raw(GGANDIVA_NODE(node)).get());
  return boost::get<Type>(gandiva_literal_node->holder());
}
```

But `LiteralNode` doesn't seem to be published.

```cpp
../gandiva-glib/node.cpp:36:27: error: no type named 'LiteralNode' in namespace 'gandiva'
    dynamic_cast<gandiva::LiteralNode*>(ggandiva_node_get_raw(GGANDIVA_NODE(node)).get());
```

Reply:
@pravindra C GLib Gandiva bindings want to use `LiteralNode::holder()` to access raw literal data. Is it acceptable that we make `gandiva/node.h` public?

Other methods in `node.h` will be useful in bindings. For example, `Node::ToString()` is useful to inspect the node object.

Reply:
sorry, I missed this. I'm fine with making gandiva/node.h public

Reply:
@pravindra Thanks! I’ll create the pull request to make `gandiva/node.h` public.

Reply:
I've created #3135.
If #3135 has been merged, I’ll rebase this branch on master.

Reply:
I've rebased on master.